### PR TITLE
improve security_group testcase use common method

### PR DESCRIPTION
**Pull request by ddaddaddda**

### CODE REVIEW

File: alicloud/resource_alicloud_security_group_test.go [hunk 1]
```diff
@@ -98,104 +97,64 @@ func testSweepSecurityGroups(region string) error {
 	return nil
 }
 
-func TestAccAlicloudSecurityGroup_basic(t *testing.T) {
-	var sg ecs.DescribeSecurityGroupAttributeResponse
+func testAccCheckSecurityGroupExists(n string, sg *ecs.DescribeSecurityGroupAttributeResponse) resource.TestCheckFunc {
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_security_group_test.go]: I think the method `testAccCheckSecurityGroupExists ` should be removed.

File: alicloud/resource_alicloud_security_group_test.go [hunk 2]
```diff
@@ -315,215 +232,160 @@ func TestAccAlicloudSecurityGroup_inner_access_2(t *testing.T) {
...
+
+
+resource "alicloud_vpc" "default" {
+  name = "${var.name}_vpc"
+  cidr_block = "10.1.0.0/21"
 }
 
-resource "alicloud_vpc" "vpc" {
+resource "alicloud_security_group" "default" {
+  count = 3
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_security_group_test.go]: Please use count = 10

File: alicloud/resource_alicloud_security_group_rule_test.go [hunk 3]
```diff
@@ -14,735 +9,199 @@ import (
...
-`
-
-const testAccSecurityGroupRuleIngress_Withnic_type = `
-variable "name" {
-  default = "tf-testAccSecurityGroupRuleVpcIngress"
-}
-resource "alicloud_security_group" "foo" {
-  vpc_id = "${alicloud_vpc.vpc.id}"
+resource "alicloud_security_group" "default" {
+  count = 2
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_security_group_rule_test.go]: Why there is `count=2`

File: alicloud/service_alicloud_ecs.go [hunk 4]
```diff
@@ -185,32 +185,42 @@ func (s *EcsService) DescribeSecurityGroupAttribute(securityGroupId string) (gro
...
-	args.NicType = nicType
+func (s *EcsService) DescribeSecurityGroupRule(id string) (rule ecs.Permission, err error) {
+	parts, err := ParseResourceId(id, 8)
+	if err != nil {
+		return rule, WrapError(err)
+	}
+	groupId, direction, ipProtocol, portRange, nicType, cidr_ip, policy := parts[0], parts[1], parts[2], parts[3], parts[4], parts[5], parts[6]
+	priority, err := strconv.Atoi(parts[7])
+	if err != nil {
+		return rule, WrapError(fmt.Errorf("parsing security_group_rule attribute 'priority' occurred an error：%s ", err.Error()))
```
Comment (xiaozhu36) [inline alicloud/service_alicloud_ecs.go]: line 204 should be `return rule, WrapError(err)`

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,11 +3,16 @@
...
+- **New Data Source:** `alicloud_snapshots` [GH-988]
 
 IMPROVEMENTS:
 
+- improve security_group testcase use common method [GH-995]
+- fix vpn change local_subnet and remote_subnet bug [GH-994]
+- fix a bug in vpn_gateway resource [GH-993]
 - improve disk dataSource testcase use common method [GH-990]
 - fix(nas): use new sdk [GH-984]
+- Feature/slb listener redirect http to https [GH-981]
```
Comment (xiaozhu36) [inline CHANGELOG.md]: Please rebase

File: alicloud/resource_alicloud_security_group_test.go (hunk 1 above)
Comment (ddaddaddda) [reply alicloud/resource_alicloud_security_group_test.go]: this method is referenced by db_instance and fc_service testcase, so hold it temporarily.

**Pull request merged**